### PR TITLE
Dataflow: Make use of defaults for language-specific hooks.

### PR DESCRIPTION
**Pull request by aschackmull**
With default predicates we can get rid of some dummy hook implementations.

### CODE REVIEW

**Review by owen-mc — APPROVED** [review]
👍🏻 Go

**Review by geoffw0 — DISMISSED** [review]
Swift :+1:

File: shared/dataflow/codeql/dataflow/DataFlow.qll [hunk 1]
```diff
@@ -226,9 +226,13 @@ signature module InputSig {
    *
    * Argument `arg` is part of a path from a source to a sink, and `p` is the target parameter.
    */
-  int getAdditionalFlowIntoCallNodeTerm(ArgumentNode arg, ParameterNode p);
+  default int getAdditionalFlowIntoCallNodeTerm(ArgumentNode arg, ParameterNode p) { none() }
 
-  predicate golangSpecificParamArgFilter(DataFlowCall call, ParameterNode p, ArgumentNode arg);
+  default predicate golangSpecificParamArgFilter(
```
Comment (hvitved) [inline shared/dataflow/codeql/dataflow/DataFlow.qll:231]: Doesn't this need a bindingset?

Comment (aschackmull) [reply shared/dataflow/codeql/dataflow/DataFlow.qll:231]: Yes, looks like it does. Will fix.

**Review by hvitved — APPROVED** [review]

**Pull request merged**